shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -578,6 +593,16 @@ - (void)dispatchKeyEvent:(NSEvent*)event ofType:(NSString*)type {
   [_keyEventChannel sendMessage:keyMessage reply:replyHandler];
 }
 
+- (void)onAccessibilityStatusChanged:(NSNotification*)notification {
+  if (!_engine) {
+    return;
+  }
+  if (@available(macOS 10.13, *)) {
+    NSWorkspace* ws = [NSWorkspace sharedWorkspace];
+    [_engine updateSemanticsEnabled:ws.voiceOverEnabled];
+  }
```

Comment:
I'm confused; if you're listening for the private notification anyway, why are you only doing anything with it on 10.13? Surely the notification has the status?

Reply:
I will revisit this. Last time i check, ws.voiceOverEnabled is only available after 10.13. There is not public API to determine whether voice over is turned on before 10.13. I need to check chromium to see how they determine the voiceover is turned on pre 10.13, but i guess it is probably another private API

Reply:
and I think listening to private notification(NSString) should not get us to be banned from app store I assume?

Reply:
> There is not public API to determine whether voice over is turned on before 10.13.

Usually a change notification has the status of the thing that changed, which is why I asked about the notification. Have you inspected its `userInfo`?

Reply:
> and I think listening to private notification(NSString) should not get us to be banned from app store I assume?

Since listening is passive (i.e., nothing breaks if it changes, unlike calling private API), I would expect not. There's no way to know definitively other than seeing if any issues arise.